### ISSUE TITLE
使用pdx.det.YOLOv3报错TypeError: 'NoneType' object is not iterable

### ISSUE THREAD

**Issue by ulen2000** [PaddlePaddle/PaddleX#1588]
使用pdx.det.YOLOv3训练数据集时报错TypeError: 'NoneType' object is not iterable

训练代码如下：

import paddlex as pdx
import os
os.environ['CUDA_VISIBLE_DEVICES'] = '3' 
#export CUDA_VISIBLE_DEVICES=3

from paddlex import transforms as T

train_transforms = T.Compose([
    T.MixupImage(mixup_epoch=250), T.RandomDistort(),
    T.RandomExpand(im_padding_value=[123.675, 116.28, 103.53]), T.RandomCrop(),
    T.RandomHorizontalFlip(), T.BatchRandomResize(
        target_sizes=[320, 352, 384, 416, 448, 480, 512, 544, 576, 608],
        interp='RANDOM'), T.Normalize(
            mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
])

eval_transforms = T.Compose([
    T.Resize(
        608, interp='CUBIC'), T.Normalize(
            mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
])

# 下载和解压表计检测数据集，如果已经预先下载，可注释掉下面两行
meter_det_dataset = 'https://bj.bcebos.com/paddlex/examples/meter_reader/datasets/meter_det.tar.gz'
pdx.utils.download_and_decompress(meter_det_dataset, path='./')

train_dataset = pdx.datasets.CocoDetection(
    data_dir='meter_det/train/',
    ann_file='meter_det/annotations/instance_train.json',
    transforms=train_transforms,
    shuffle=True)
eval_dataset = pdx.datasets.CocoDetection(
    data_dir='meter_det/test/',
    ann_file='meter_det/annotations/instance_test.json',
    transforms=eval_transforms)

num_classes = len(train_dataset.labels)

model=pdx.det.YOLOv3(num_classes=num_classes, backbone='MobileNetV1', anchors=None, anchor_masks=None,
                         ignore_threshold=0.7, nms_score_threshold=0.01, nms_topk=1000, nms_keep_topk=100,
                         nms_iou_threshold=0.45, label_smooth=False,
                         train_random_shapes=[320, 352, 384, 416, 448, 480, 512, 544, 576, 608], input_channel=3)

model.train(num_epochs=270,
            train_dataset=train_dataset,
            train_batch_size=8,
            eval_dataset=eval_dataset,
            save_interval_epochs=20,
            log_interval_steps=2,
            save_dir='output/yolo3_MobileNetV1',
            pretrain_weights='COCO',
            optimizer=None,
            earning_rate=1.0/8000,
            warmup_steps=1000,
            warmup_start_lr=0.0,
            lr_decay_epochs=[213, 240],
            lr_decay_gamma=0.1,
            metric=None,
            use_vdl=True,
            sensitivities_file=DEFAULT,
            eval_metric_loss=0.05,
            early_stop=True,
            early_stop_patience=5,
            resume_checkpoint=None)

报错如下图：
![选区_014](https://user-images.githubusercontent.com/58459187/189266362-e76f991d-3dca-4a49-b585-d9b6e9626e30.png)

**Comment by ucsk** [comment]
```python
model = pdx.det.YOLOv3(
    num_classes=len(train_dataset.labels),
    backbone='MobileNetV1')

model.train(
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,

    num_epochs=270,
    train_batch_size=8,
    learning_rate=0.000125,
    warmup_steps=500,
    warmup_start_lr=0.0,
    lr_decay_epochs=[213, 240],
    lr_decay_gamma=0.1,

    save_interval_epochs=5,
    log_interval_steps=20,
    save_dir='output/YOLOv3_MobileNetV1',
    pretrain_weights='COCO',

    use_vdl=True,
    early_stop=True,
    early_stop_patience=5)
```

**Comment by ulen2000** [comment]
> 

谢谢，可以了，不过能问一下本来的问题出在哪了吗

**Comment by lailuboy** [comment]
YOLO需要设置anchors，不能为None，或者使用默认值。
